sdk/eventhub/Azure.Messaging.EventHubs/src/Authorization/EventHubTokenCredential.cs

```diff
@@ -0,0 +1,72 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Messaging.EventHubs.Core;
+
+namespace Azure.Messaging.EventHubs.Authorization
+{
+    /// <summary>
+    ///   Provides a generic token-based credential for a given Event Hub instance.
+    /// </summary>
+    ///
+    /// <seealso cref="Azure.Core.TokenCredential" />
+    ///
+    internal class EventHubTokenCredential : TokenCredential
+    {
+        /// <summary>
+        ///   The Event Hubs resource to which the token is intended to serve as authorization.
+        /// </summary>
+        ///
+        public string Resource { get; }
+
+        /// <summary>
+        ///   The <see cref="TokenCredential" /> that forms the basis of this security token.
+        /// </summary>
+        ///
+        private TokenCredential Credential { get; }
+
+        /// <summary>
+        ///   Initializes a new instance of the <see cref="SharedAccessSignatureCredential"/> class.
+        /// </summary>
+        ///
+        /// <param name="tokenCredential">The <see cref="TokenCredential" /> on which to base the token.</param>
+        /// <param name="eventHubResource">The Event Hubs resource to which the token is intended to serve as authorization.</param>
+        ///
+        public EventHubTokenCredential(TokenCredential tokenCredential,
```

Comment:
What scenario does EventHubTokenCredential bring which isn't covered by TokenCredential? It seems like a simple wrapper around TokenCredential. 🤔

Reply:
I also noticed that the Resource property isn't used at all.

Reply:
Never mind. I see its usage in Track1GenericTokenProvider. Ignore this.

Reply:
Yup, it's a wrapper for the compatibility shim.   I'm hoping to be able to kill this type post-preview and use the base generic token.

Reply:
I had the same question!! Glad I expanded this collapsed conversation :)